consensus/dummy/dynamic_fee_window.go

```diff
@@ -0,0 +1,92 @@
+// (c) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dummy
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+var ErrInsufficientDynamicFeeWindowLength = errors.New("insufficient length for dynamic fee window")
+
+// DynamicFeeWindow is a window of the last [params.RollupWindow] seconds of gas
+// usage.
+//
+// Index 0 is the oldest entry, and [params.RollupWindow]-1 is the current
+// entry.
+type DynamicFeeWindow [params.RollupWindow]uint64
+
+func ParseDynamicFeeWindow(bytes []byte) (DynamicFeeWindow, error) {
+	if len(bytes) < params.DynamicFeeExtraDataSize {
+		return DynamicFeeWindow{}, fmt.Errorf("%w expected at least %d, but found %d",
+			ErrInsufficientDynamicFeeWindowLength,
+			params.DynamicFeeExtraDataSize,
+			len(bytes),
+		)
+	}
+
+	var window DynamicFeeWindow
```

Comment:
supernit you could save a line if using named returns

```suggestion
func ParseDynamicFeeWindow(bytes []byte) (window DynamicFeeWindow, err error) {
	if len(bytes) < params.DynamicFeeExtraDataSize {
		return DynamicFeeWindow{}, fmt.Errorf("%w expected at least %d, but found %d",
			ErrInsufficientDynamicFeeWindowLength,
			params.DynamicFeeExtraDataSize,
			len(bytes),
		)
	}

```

Reply:
I typically avoid using named returns other than for exceptional cases... It doesn't seem like named returns are very common in coreth, so I think it makes sense to keep the line rather than using named returns.